doc/guides/logging/dictionary.rst

```diff
@@ -0,0 +1,208 @@
+
+            memset(vs0, 0, sizeof(vs0));
+            snprintk(&vs0[0], sizeof(vs0), "%s", "dynamic str");
+
+            memset(vs1, 0, sizeof(vs1));
+            snprintk(&vs1[0], sizeof(vs1), "%s", "another dynamic str");
+
+            LOG_DBG("char %c", c);
+            LOG_DBG("s str %s", s);
+            LOG_DBG("d str %s", vs0);
+            LOG_DBG("mixed str %s %s %s %s %s %s %s", vs0, "---", vs0, "---", vs1, "---", vs1);
+            LOG_DBG("mixed c/s %c %s %s %s %c", c, s, vs0, s, c);
+
+            void *p = main;
+
+            LOG_DBG("pointer %p", p);
+
+            LOG_HEXDUMP_DBG(hexdump_msg, strlen(hexdump_msg), "For HeXdUmP!");
+
+      #ifdef CONFIG_FPU
+            float f = 66.67;
+            double d = 68.69;
+            long double ld = 70.71;
+
+            LOG_DBG("float %f, double %f, long double %Lf", f, d, ld);
+      #endif
+      }
+
+
+2. Modify :file:`samples/hello_world/src/prj.conf` to enable UART backend
+   for dictionary-based logging, and to disable text-based UART backend.
+
+   .. code-block:: none
+
+      CONFIG_LOG=y
+      CONFIG_LOG_PRINTK=y
+      CONFIG_LOG_DICTIONARY_BACKEND_UART=y
+      CONFIG_LOG_BACKEND_UART=n
+      CONFIG_LOG2_MODE_IMMEDIATE=y
+
+3. Build and run the `hello_world` application:
+
+   .. code-block:: console
+
+      west build -d /tmp/build -p -b qemu_x86 samples/hello_world/ -t run
+
+4. Here is the output from QEMU:
+
+   .. code-block:: none
+
+      SeaBIOS (version zephyr-v1.0.0-0-g31d4e0e-dirty-20200714_234759-fv-az50-zephyr)
+      Booting from ROM..000004000000000000a9d3000004000000e05810001e6010009f5f10000000030000000000001309010003000000c4511000d45e1000000802000000000000fc2f010002000000b45110000020030000000000003c51010003000000a05110005c5b10000018020000000000001378010002000000945110000020050000000000005099010005000000785110005c5b1000010000000200000000200500000000000069cb0100050000005c5110005c5b1000100000001100000000200500000000000096fd010005000000405110005c5b10002000000021000000002007000000000000c32f020007000000205110005c5b100040000000000000004100000000000000002004000000000000e689020004000000045110005c5b100021000000002004000000000000fcbc020004000000f45010005c5b1000f45e100000600700000000000073f3020004010000e45010005c5b1000000000000364796e616d6963207374720000201b000000000000c35003000a040000c05010005c5b100000000000ff5e100000000000ff5e100000000000ff5e1000000000000364796e616d696320737472000564796e616d6963207374720007616e6f746865722064796e616d6963207374720009616e6f746865722064796e616d6963207374720000600b000000000000260f040008010000845010005c5b100021000000f45e100000000000f45e1000210000000564796e616d696320737472000020040000000000009064040004000000745010005c5b1000da0310000020041a0000000000209a0400040000006c5010005c5b10001e5f100048455844554d50212048455844554d50402048455844554d502300200a00000000000000ec04000a000000405010005c5b100000000040e1aa50405c8fc2f5282c514085eb51b81e856b8d05400000
+
+5. Copy the hexidecimal string into a temporary file ``/tmp/serial.log``:
```

Comment:
Copying the log to temp file sounds quite cumbersome. Could the output be piped directly to log_parser.py instead?

Reply:
Piping the output from west build or west run sends all data to the parser. There is no easy way to distinguish between normal text and log data from a stream of text. Though output QEMU, the log would be captured through serial or flash storage which would only contain log binary data. So I am not sure if we want to make the log parser more complicated so it could parse the build output.

Reply:
Sure, I understand. It is just that copying data from console to temp file is not very practical in real life. Could the log parser just monitor some start bytes/marker and ignore all the data it receives before that. That sounds like an easy solution to this issue, what do you think?

Reply:
Added a separator so the parser can find where the log begins.